### DISCUSSION TITLE
3.2.0-rc1

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-studio#225] [Announcements]
## What's Changed
* Add Persistence strategy at Workflow / Activity Level by @jdevillard in https://github.com/elsa-workflows/elsa-studio/pull/136
* fixed bug where you were allowed to connect to an out port by @cr0fters in https://github.com/elsa-workflows/elsa-studio/pull/195
* fixed bug where you were allowed to start a connection from an in port by @cr0fters in https://github.com/elsa-workflows/elsa-studio/pull/198
* Javascript Intellisense by @sNakiex in https://github.com/elsa-workflows/elsa-studio/pull/200
* Expose additional workflow definition editor events by @sfmskywalker in https://github.com/elsa-workflows/elsa-studio/pull/213
* Global read-only and read-only workflows by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-studio/pull/203
* Made checkboxes disabled in readonly mode by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-studio/pull/215
* Fix loading of workflow graph by @sfmskywalker in https://github.com/elsa-workflows/elsa-studio/pull/218
* Allow user to use the history tab if older version is selected by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-studio/pull/220
* Update WorkflowDefinitionService and Elsa.Api.Client version by @raymonddenhaan in https://github.com/elsa-workflows/elsa-studio/pull/221
* Refactor activity execution service method parameters by @sfmskywalker in https://github.com/elsa-workflows/elsa-studio/pull/222
* Update package versions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-studio/pull/223
* Update ELSA client by @raymonddenhaan in https://github.com/elsa-workflows/elsa-studio/pull/224

## New Contributors
* @cr0fters made their first contribution in https://github.com/elsa-workflows/elsa-studio/pull/195

**Full Changelog**: https://github.com/elsa-workflows/elsa-studio/compare/3.1.0...3.2.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-studio/releases/tag/3.2.0-rc1'>3.2.0-rc1</a>.</em>

**Comment by glime-ai** [comment]
The changes between Elsa Studio version 3.1.0 and 3.2.0-rc1 include several bug fixes, feature additions, and updates. Here's a summary of the key changes:

1. **Persistence Strategy at Workflow/Activity Level**: A new feature that allows setting a persistence strategy at the workflow or activity level. (PR [#136](https://github.com/elsa-workflows/elsa-studio/pull/136))

2. **Bug Fixes**: Several bugs were fixed, including issues with connecting ports incorrectly in the workflow designer. (PRs [#195](https://github.com/elsa-workflows/elsa-studio/pull/195), [#198](https://github.com/elsa-workflows/elsa-studio/pull/198))

3. **Javascript Intellisense**: Added Intellisense support for JavaScript expressions, improving the developer experience when writing scripts within the workflow designer. (PR [#200](https://github.com/elsa-workflows/elsa-studio/pull/200))

4. **Expose Additional Workflow Definition Editor Events**: Additional events have been exposed to enhance the interaction with the workflow editor component. (PR [#213](https://github.com/elsa-workflows/elsa-studio/pull/213))

5. **Global Read-Only and Read-Only Workflows**: Introduced the ability to set workflows as read-only globally or on a per-workflow basis. (PR [#203](https://github.com/elsa-workflows/elsa-studio/pull/203))

6. **UI Enhancements**: Made checkboxes disabled in read-only mode to prevent changes. (PR [#215](https://github.com/elsa-workflows/elsa-studio/pull/215))

7. **Workflow Graph Loading Fix**: Addressed an issue with loading the workflow graph. (PR [#218](https://github.com/elsa-workflows/elsa-studio/pull/218))

8. **History Tab Usability**: Allowed users to use the history tab even if an older version of a workflow is selected. (PR [#220](https://github.com/elsa-workflows/elsa-studio/pull/220))

9. **Service and Client Updates**: Updated the WorkflowDefinitionService and Elsa.Api.Client versions. (PRs [#221](https://github.com/elsa-workflows/elsa-studio/pull/221), [#222](https://github.com/elsa-workflows/elsa-studio/pull/222), [#223](https://github.com/elsa-workflows/elsa-studio/pull/223), [#224](https://github.com/elsa-workflows/elsa-studio/pull/224))

10. **New Contributor**: @cr0fters made their first contribution. (PR [#195](https://github.com/elsa-workflows/elsa-studio/pull/195))

The full changelog can be found at the provided link: [3.1.0...3.2.0-rc1 Changelog](https://github.com/elsa-workflows/elsa-studio/compare/3.1.0...3.2.0-rc1).

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
